src/write_gcm.c

```diff
@@ -2419,16 +2419,17 @@ static char *wg_get_from_metadata_server(const char *base, const char *resource,
     return NULL;
   }
 
-  char buffer[2048];
-  if (wg_curl_get_or_post(buffer, sizeof(buffer), url, NULL, headers,
-      num_headers, silent_failures) != 0) {
+  char *response = NULL;
+  if (wg_curl_get_or_post(&response, url, NULL, headers, num_headers,
+      silent_failures) != 0) {
+    sfree(response);
     if (!silent_failures) {
       ERROR("write_gcm: wg_get_from_metadata_server failed to fetch metadata"
             "from %s", url);
```

Comment:
Minor drive-by fix: missing space between `metadata` and `from`.

Reply:
[Really optional] It's usually better to put such spaces at the start of the next line, so they're hard to miss.